### PR TITLE
Add k3s etcd restoration test

### PR DESCRIPTION
**Pull request by galal-hussein**
Signed-off-by: galal-hussein <hussein.galal.ahmed.11@gmail.com>

#### Proposed Changes ####

Adding k3s snapshot restoration integration test

### CODE REVIEW

File: tests/util/unit.go [hunk 1]
```diff
@@ -69,3 +70,20 @@ func ServiceIPNet() *net.IPNet {
 	_, serviceIPNet, _ := net.ParseCIDR("10.43.0.0/16")
 	return serviceIPNet
 }
+
+func FindFile(root, findStr string) (string, error) {
```
Comment (briandowns) [inline tests/util/unit.go:74]:
```suggestion
func FindFile(root, filename string) (string, error) {
```

File: tests/util/integration.go [hunk 2]
```diff
@@ -146,9 +149,9 @@ func K3sStartServer(inputArgs ...string) (*K3sServer, error) {
 	for _, arg := range inputArgs {
 		cmdArgs = append(cmdArgs, strings.Fields(arg)...)
 	}
-
+	fmt.Println("cmd args: %#v", cmdArgs)
```
Comment (briandowns) [inline tests/util/integration.go:152]: Where these fmt lines intended or debug?

File: tests/util/integration.go [hunk 3]
```diff
@@ -201,3 +204,16 @@ func K3sCleanup(server *K3sServer, releaseLock bool) error {
 	}
 	return nil
 }
+
+// RunCommand Runs command on the cluster accessing the cluster through kubeconfig file
+func RunCommand(cmd string) (string, error) {
+	c := exec.Command("bash", "-c", cmd)
+	time.Sleep(10 * time.Second)
```
Comment (dereknola) [inline tests/util/integration.go:211]: There should be a need to sleep when running bash commands
```suggestion
```

**Review by dereknola — CHANGES_REQUESTED** [review]

File: pkg/etcd/int_tests/restore/etcd_restore_int_test.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+			// get snapshot file
+			filePath, err := testutil.RunCommand(`sudo find ` + tmpdDataDir + `/server -name "*snapshot-to-restore*"`)
+			Expect(err).ToNot(HaveOccurred())
+			filePath = strings.TrimSuffix(filePath, "\n")
+			Eventually(func() (string, error) {
+				return testutil.K3sCmd("server", "-d", tmpdDataDir, "--cluster-reset", "--token", "test", "--cluster-reset-restore-path", filePath)
+			}, "360s", "5s").Should(ContainSubstring(`Etcd is running, restart without --cluster-reset flag now`))
+			// out, _ :=
+			// // Expect(err).ToNot(HaveOccurred())
+			// Expect(out).To()
```
Comment (dereknola) [inline pkg/etcd/int_tests/restore/etcd_restore_int_test.go:75]: Did you still need these?

File: tests/util/unit.go [hunk 5]
```diff
@@ -69,3 +70,20 @@ func ServiceIPNet() *net.IPNet {
 	_, serviceIPNet, _ := net.ParseCIDR("10.43.0.0/16")
 	return serviceIPNet
 }
+
+func FindFile(root, filename string) (string, error) {
```
Comment (dereknola) [inline tests/util/unit.go:74]: It doesn't look like this is being used anywhere, do we still need it?

**Review by dereknola — CHANGES_REQUESTED** [review]

**Review by briandowns — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

**Pull request merged**